### ISSUE TITLE
Issue on new install -- not saving settings

### ISSUE THREAD

**Issue by codefaux** [home-assistant/android#2217]
**Home Assistant Android version:**
2021.12.3-full
2022.1.1-full

**Android version:**
Android 11 (LineageOS 18.1 nightly)

**Phone model:**
HD1907

**Home Assistant version:**
2021.10.5

**Last working Home Assistant release (if known):**
Not relevant

**Description of problem:**
Our phones continue to log in and store instance information, but on my phone the Power Menu "Device Controls" stopped working a few weeks (ish?) ago. I assumed this was irrelevant at the time, but may have been early symptoms.
We just purchased and set up two new phones, and on each of them the behavior is identical. We have, for unrelated reasons, installed the app on several different fresh re-installs of LineageOS, each time resulting in the same behavior.
The Home Assistant Companion app loads, requires manual URL entry, allows login, but fails to store any information. Upon closing and re-opening the app, it requires full setup (instance, login, device name/tracking) every time. When loaded, the app shows no name configured under the Device Name entry for Configuration, so I believe the issue presents immediately upon any attempted user data write.

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**
```
--------- beginning of main
01-30 17:30:42.632 20786 20786 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
01-30 17:30:42.632 20786 20786 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
01-30 17:30:42.657 20786 20786 I FirebaseApp: Device unlocked: initializing all Firebase APIs for app [DEFAULT]
01-30 17:30:42.668 20786 20786 I FirebaseInitProvider: FirebaseApp initialization successful
01-30 17:30:42.672 20786 20786 D WM-WrkMgrInitializer: Initializing WorkManager with default configuration.
01-30 17:30:42.676 20786 20786 I TetheringManager: registerTetheringEventCallback:io.homeassistant.companion.android
01-30 17:30:42.685 20786 21877 I DynamiteModule: Considering local module com.google.android.gms.measurement.dynamite:55 and remote module com.google.android.gms.measurement.dynamite:59
01-30 17:30:42.685 20786 21877 I DynamiteModule: Selected remote version of com.google.android.gms.measurement.dynamite, version >= 59
01-30 17:30:42.685 20786 21877 V DynamiteModule: Dynamite loader version >= 2, using loadModule2NoCrashUtils
01-30 17:30:42.687 20786 20786 D Sentry  : io.sentry.auto-init read: false
01-30 17:30:42.687 20786 20786 I Sentry  : Retrieving auto-init from AndroidManifest.xml
01-30 17:30:42.711 20786 20786 W pool-13-thread-: type=1400 audit(0.0:19420): avc: denied { read } for name="version" dev="proc" ino=4026532146 scontext=u:r:untrusted_app:s0:c240,c256,c512,c768 tcontext=u:object_r:proc_version:s0 tclass=file permissive=0 app=io.homeassistant.companion.android
01-30 17:30:42.742 20786 21899 I FA      : App measurement initialized, version: 50018
01-30 17:30:42.742 20786 21899 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
01-30 17:30:42.743 20786 21899 I FA      : To enable faster debug mode event logging run:
01-30 17:30:42.743 20786 21899 I FA      :   adb shell setprop debug.firebase.analytics.app io.homeassistant.companion.android
01-30 17:30:42.797 20786 20786 I WebViewFactory: Loading com.google.android.webview version 97.0.4692.98 (code 469209834)
01-30 17:30:42.814 20786 20786 I cr_WVCFactoryProvider: Loaded version=97.0.4692.98 minSdkVersion=29 isBundle=true multiprocess=true packageId=2
01-30 17:30:42.820 20786 21913 I cr_VariationsUtils: Failed reading seed file "/data/user/0/io.homeassistant.companion.android/app_webview/variations_seed_new"
01-30 17:30:42.822 20786 20786 I cr_LibraryLoader: Successfully loaded native library
01-30 17:30:42.823 20786 20786 I cr_CachingUmaRecorder: Flushed 8 samples from 8 histograms.
01-30 17:30:42.848 20786 21899 I FA      : Tag Manager is not found and thus will not be used
01-30 17:30:42.883 20786 20786 W mpanion.androi: Accessing hidden method Landroid/content/res/Resources$Theme;->getKey()Landroid/content/res/Resources$ThemeKey; (greylist-max-o, reflection, denied)
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: QUALCOMM build                   : 559e78c, Ib9d997268a
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: Build Date                       : 10/13/20
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.32.02.02
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: Local Branch                     : mybrancheb0d76cb-f27a-f1ff-9cdc-1458d0b52ae8
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: Remote Branch                    : quic/gfx-adreno.lnx.1.0.r99-rel
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: Remote Branch                    : NONE
01-30 17:30:42.908 20786 21889 I AdrenoGLES-0: Reconstruct Branch               : NOTHING
01-30 17:30:42.909 20786 21889 I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch64
01-30 17:30:42.909 20786 21889 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
01-30 17:30:42.914 20786 21889 I AdrenoGLES-0: PFP: 0x016ee190, ME: 0x00000000
01-30 17:30:42.949 20786 20786 D SensorReceiver: Sensor audio_sensor corresponding to received event android.media.RINGER_MODE_CHANGED is disabled, skipping sensors update
01-30 17:30:42.950 20786 21881 W SensorReceiver: Device not registered, skipping sensor update/registration
01-30 17:30:42.952 20786 20786 D UrlRepository: localUrl is: false and usesInternalSsid is: false
01-30 17:30:42.952 20786 20786 D UrlRepository: Using external URL
01-30 17:30:43.006 20786 20786 W mpanion.androi: Class androidx.compose.runtime.snapshots.SnapshotStateList failed lock verification and will run slower.
01-30 17:30:43.006 20786 20786 W mpanion.androi: Common causes for lock verification issues are non-optimized dex code
01-30 17:30:43.006 20786 20786 W mpanion.androi: and incorrect proguard optimizations.
01-30 17:30:44.123 20786 21921 D HomeAssistantSearcher: Service discovery started
01-30 17:30:44.998 20786 21921 I HomeAssistantSearcher: Discovery stopped: _home-assistant._tcp
01-30 17:37:33.483 20786 21921 D HomeAssistantSearcher: Service discovery started
01-30 17:37:34.686 20786 21921 I HomeAssistantSearcher: Discovery stopped: _home-assistant._tcp
01-30 17:37:35.697 20786 20786 D Compose Focus: Owner FocusChanged(true)
01-30 17:37:35.774 20786 20786 W IInputConnectionWrapper: beginBatchEdit on inactive InputConnection
01-30 17:37:35.774 20786 20786 W IInputConnectionWrapper: endBatchEdit on inactive InputConnection
01-30 17:37:35.774 20786 20786 W IInputConnectionWrapper: beginBatchEdit on inactive InputConnection
01-30 17:37:35.774 20786 20786 W IInputConnectionWrapper: endBatchEdit on inactive InputConnection
01-30 17:37:46.884 20786 20786 W Choreographer: Frame time is 0.020487 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
01-30 17:37:49.711 20786 20786 E DecorView[]: Destroying unexpected ActionMode instance of TYPE_FLOATING; com.android.internal.view.FloatingActionMode@b29dc53 was not the current floating action mode! Expected null
01-30 17:37:49.843 20786 21889 D OpenGLRenderer: endAllActiveAnimators on 0xb400007991578250 (RippleDrawable) with handle 0xb4000078b1513da0
01-30 17:37:51.042 20786 20786 I cr_VariationsUtils: Loaded seed with age 7257s
01-30 17:37:51.112 20786 20786 I System.out: recordForce value 5
01-30 17:37:51.143 20786 20786 D Compose Focus: Owner FocusChanged(false)
01-30 17:37:51.425 20786 20786 I chromium: [INFO:CONSOLE(811)] "The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path for more information.", source: https://hass.<REDACT>.org/frontend_latest/authorize.71a4a9a1.js (811)
01-30 17:38:02.191 20786 20786 W Choreographer: Frame time is 0.174621 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
01-30 17:38:06.694 20786 20786 D UrlRepository: localUrl is: false and usesInternalSsid is: false
01-30 17:38:06.764 20786 22193 D UrlRepository: localUrl is: false and usesInternalSsid is: false
01-30 17:38:06.764 20786 22193 D UrlRepository: Using external URL
01-30 17:38:06.963 20786 22201 D UrlRepository: localUrl is: false and usesInternalSsid is: false
01-30 17:38:06.963 20786 22201 D UrlRepository: Using external URL
01-30 17:38:07.038 20786 20786 E IntegrationRepository: Unable to register device
01-30 17:38:07.038 20786 20786 E IntegrationRepository: retrofit2.HttpException: HTTP 404 
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:161)
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
01-30 17:38:07.038 20786 20786 E IntegrationRepository: 	at java.lang.Thread.run(Thread.java:923)
01-30 17:38:07.091 20786 20786 D RenderScript HIDL Adaptation: IRenderScriptDevice::getService()
01-30 17:38:07.092 20786 20786 D RenderScript HIDL Adaptation: IRenderScriptDevice::getService() returned 0x0
01-30 17:38:07.093 20786 20786 D RenderScript HIDL Adaptation: Using Fallback Path.
01-30 17:38:07.099 20786 20786 D mpanion.androi: Successfully queried cache dir: /data/user_de/0/io.homeassistant.companion.android/code_cache
01-30 17:38:07.099 20786 20786 D RenderScript: Setting cache dir: /data/user_de/0/io.homeassistant.companion.android/code_cache
01-30 17:38:07.102 20786 20786 I System.out: recordForce value 5
01-30 17:38:07.103 20786 20786 D WebviewActivity: Current webview package com.google.android.webview and version 97.0.4692.98
01-30 17:38:07.163 20786 21881 D SensorWorker: Updating all Sensors.
01-30 17:38:07.171 20786 20786 D UrlRepository: localUrl is: false and usesInternalSsid is: false
01-30 17:38:07.171 20786 20786 D UrlRepository: Using external URL
01-30 17:38:07.172 20786 22217 I WM-Processor: Moving WorkSpec (20e5f4c6-1741-410c-8571-8ad897a42ddd) to the foreground
01-30 17:38:07.177 20786 20786 I WM-SystemFgDispatcher: Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android/androidx.work.impl.foreground.SystemForegroundService (has extras) }
01-30 17:38:07.186 20786 21881 W SensorReceiver: Device not registered, skipping sensor update/registration
01-30 17:38:07.188 20786 21878 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=20e5f4c6-1741-410c-8571-8ad897a42ddd, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
01-30 17:38:07.199 20786 20786 I WM-SystemFgDispatcher: Stopping foreground service
01-30 17:38:07.236 20786 20786 I chromium: [INFO:CONSOLE(1277)] "The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path for more information.", source: https://hass.<REDACT>.org/frontend_latest/app.d030393d.js (1277)
01-30 17:38:07.504 20786 22220 D WebviewActivity: External bus {"type":"connection-status","payload":{"event":"connected"},"id":1}
01-30 17:38:07.615 20786 22220 D WebviewActivity: External bus {"type":"config/get","id":2}
01-30 17:38:07.616 20786 20786 D WebviewActivity: externalBus({"id":2,"type":"result","success":true,"result":{"hasSettingsScreen":true,"canWriteTag":true,"hasExoPlayer":true}});
01-30 17:38:07.637 20786 20786 D WebviewActivity: Callback null
01-30 17:38:07.639 20786 20786 D WebviewActivity: Color from webview is "#101e24-SPACER-#111111"
01-30 17:38:07.639 20786 20786 D WebviewActivity: Color from webview is "#101e24-SPACER-#111111"
01-30 17:38:07.640 20786 21881 D WebViewPresenterImpl: Try getting color from webview color "#101e24".
01-30 17:38:07.640 20786 21881 I WebViewPresenterImpl: Found color -15720924.
01-30 17:38:07.641 20786 21881 D WebViewPresenterImpl: Try getting color from webview color "#111111".
01-30 17:38:07.641 20786 21881 I WebViewPresenterImpl: Found color -15658735.
01-30 17:38:11.207 20786 22220 D WebviewActivity: External bus {"type":"config_screen/show","id":3}
01-30 17:38:11.260 20786 20786 D UrlRepository: Using external URL
01-30 17:38:11.261 20786 20786 I chatty  : uid=10240(io.homeassistant.companion.android) identical 2 lines
01-30 17:38:11.262 20786 20786 D UrlRepository: Using external URL
01-30 17:38:11.415 20786 21881 D SensorWorker: Updating all Sensors.
01-30 17:38:11.417 20786 21878 I WM-Processor: Moving WorkSpec (4019779f-230b-4ad6-8fb6-ebdc3b28a81f) to the foreground
01-30 17:38:11.425 20786 21882 W SensorReceiver: Device not registered, skipping sensor update/registration
01-30 17:38:11.426 20786 20786 I WM-SystemFgDispatcher: Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android/androidx.work.impl.foreground.SystemForegroundService (has extras) }
01-30 17:38:11.427 20786 21885 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=4019779f-230b-4ad6-8fb6-ebdc3b28a81f, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
01-30 17:38:11.441 20786 20786 I WM-SystemFgDispatcher: Stopping foreground service
01-30 17:38:11.781 20786 21881 E IntegrationRepository: Unable to get notification rate limits
01-30 17:38:11.781 20786 21881 E IntegrationRepository: retrofit2.HttpException: HTTP 403 
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:161)
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
01-30 17:38:11.781 20786 21881 E IntegrationRepository: 	at java.lang.Thread.run(Thread.java:923)
01-30 17:38:11.782 20786 21881 D SettingsPresenter: Unable to get rate limits
01-30 17:38:12.995 20786 20786 W Choreographer: Frame time is 0.118885 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
01-30 17:38:14.425 20786 21881 D LogcatReader: Read logcat for pid 20786
01-30 17:38:14.440 20786 21881 D LogcatReader: Done reading logcat for pid 20786
01-30 17:38:18.264 20786 21881 D LogcatReader: Read logcat for pid 20786
```

**Screenshot of problem:**
![image](https://user-images.githubusercontent.com/42219549/151727905-186b3aaf-192a-4ba1-80dc-d344d194a825.png)

**Additional information:**

**Comment by dshokouhi** [comment]
Can you double check that you gave either `default_config` or `mobile_app` in your configuration.yaml file? Sounds like you don't have that.

**Comment by codefaux** [comment]
Your comment led me to both a long-needed update and a bit of a cleanup of a disasterous config file. It's still a mess but the app and the site both work now. Fascinating that old logged in devices worked and fresh logins refused to store data. This is resolved BUT could you provide some insight to that? I'm deeply curious. Much appreciated, thanks!

**Comment by dshokouhi** [comment]
> Fascinating that old logged in devices worked and fresh logins refused to store data. This is resolved BUT could you provide some insight to that?

Because if you had those config options previously then it was able to get registered, then later on when it was no longer there you could no longer register and got a 404 error as the API was no longer accessible due to the missing config.

**Comment by dshokouhi** [comment]
Leaving this issue open while we try to see if there is a better way to inform a user. Unfortunately getting the config response requires the app to be registered so we may opt for a generic error message here.  Given the symptoms and that the app can't fix this issue we should probably add a troubleshooting step for users who hit this issue.

**Comment by codefaux** [comment]
> > Fascinating that old logged in devices worked and fresh logins refused to store data. This is resolved BUT could you provide some insight to that?
> 
> Because if you had those config options previously then it was able to get registered, then later on when it was no longer there you could no longer register and got a 404 error as the API was no longer accessible due to the missing config.

Understood, but to clarify, I did not change anything between the period of phones working/registering, and the period where they stopped. That config file has been unchanged since I migrated it from a HASS.IO Docker container back when you stopped providing that. -- save commenting out a single integration that stopped working a year and two months ago. Also, I did/do have a ```default_config``` section. I did not have the ```mobile_app``` section but documentation implies it was included in default_config. I added the latter during troubleshooting just to be sure. I also deleted wholesale my apparently-broken HACS and several modules from it (The HACS install was from around the HASS.IO period) and updated the whole system. The config change (mobile_app) made no impact, but the update and/or nuking old stuff did.

Phones have been able to register and stay logged in/etc until very recently (family visit, spare room lighting is automation only) though it's difficult to say how recently.  That's why I'm so confused here - I didn't change anything, it's an incredibly old and unmaintained system. To my knowledge nothing changed in the last year ish.

Very glad to hear my broken-ness might lead to a situational improvement though.
